exasol/nb_connector/extension_wrapper_common.py

```diff
@@ -101,11 +108,20 @@ def deploy_language_container(conf: Secrets,
             extract_validator=validator
         )
 
-        deployer.download_and_run(container_url,
-                                  container_name,
-                                  alter_system=False,
-                                  allow_override=allow_override,
-                                  wait_for_completion=True)
+        if container_file:
+            deployer.run(container_file,
+                         container_name,
+                         alter_system=False,
+                         allow_override=allow_override,
+                         wait_for_completion=True)
+        elif container_url:
+            deployer.download_and_run(container_url,
```

Comment:
Methods should not have an "and" in their name. This is a clear sign of broken SoC.

Reply:
There is a separate function that runs the deployment with a given container file. This function downloads the container first and then calls the run function. I wouldn't take this "and" rule religiously. I think in this case the name is alright. It's in a different repo anyway.

Reply:
Accepted. With belly grumbling.